### PR TITLE
ci(engine): update restart policy in docker compose tpl

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: ref #7380

### What is changed and how it works?

- Change docker compose restart policy to `unless-stopped`
- Check whether panic happens in master and executor logs, if it does, fail the test directly.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/test/utils/check_log.sh [hunk 1]
```diff
@@ -1,3 +1,12 @@
 #!/bin/bash
 
-grep "Run all test success" /tmp/tiflow_engine_test/engine_it.log && exit 0 || exit 1
+grep "Run all test success" /tmp/tiflow_engine_test/engine_it.log || exit 1
+
+grep -rE "\[PANIC\]|panic:" /tmp/tiflow_engine_test
+
+if [ $? -eq 0 ]; then
```
Comment (lance6716) [inline engine/test/utils/check_log.sh:7]: should be not equal 0 for "not found panic"?

**Review by lance6716 — APPROVED** [review]

File: deployments/engine/docker-compose/1m1e_with_s3.yaml [hunk 2]
```diff
@@ -42,7 +42,6 @@ services:
       - "10241:10241"
     depends_on:
       - "server-master"
-    restart: "on-failure"
```
Comment (CharlesCheung96) [inline deployments/engine/docker-compose/1m1e_with_s3.yaml:45]: Why not use `restart: unless-stopped` policy here?

Comment (amyangfei) [reply deployments/engine/docker-compose/1m1e_with_s3.yaml:45]: L49 has one.

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**